File: app/src/main/java/com/example/mobile/julfani/tubes/entity/SleepViewModel.java
package com.example.mobile.julfani.tubes.entity;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class SleepViewModel extends AndroidViewModel {
    private final SleepRepository sleepRepository;
    private final LiveData<List<Sleep>> allData;

    public SleepViewModel(@NonNull Application application) {
        super(application);
        sleepRepository = new SleepRepository(application);
        allData = sleepRepository.getAllData();
    }

    public LiveData<List<Sleep>> getAllData(){
        return allData;
    }

    public void insert(Sleep sleep){
        sleepRepository.insert(sleep);
    }

    public void deleteAll(){
        sleepRepository.deleteAll();
    }
}

File: app/src/main/java/com/nursyah/bettersleep/entity/SleepViewModel.kt
package com.nursyah.bettersleep.entity

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

class SleepViewModel(application: Application) : AndroidViewModel(application) {
    private val sleepRepository: SleepRepository
    private val allData: LiveData<List<Sleep>>

    fun getAllData(): LiveData<List<Sleep>>{
        return allData
    }

    fun insert(sleep: Sleep?) {
        sleepRepository.insert(sleep)
    }

    fun deleteAll() {
        sleepRepository.deleteAll()
    }

    init {
        sleepRepository = SleepRepository(application)
        allData = sleepRepository.getAllData()
    }
}